diff --git a/doomgeneric/i_sound.c b/doomgeneric/i_sound.c
index c80a9d1..c5afae6 100644
--- a/doomgeneric/i_sound.c
+++ b/doomgeneric/i_sound.c
@@ -18,7 +18,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 
-#if defined(FEATURE_SOUND) && !defined(__DJGPP__)
+#if defined(FEATURE_SOUND) && !defined(__DJGPP__) && !defined(FEATURE_SOUND_GODOT)
 #include <SDL_mixer.h>
 #endif
 
@@ -408,7 +408,7 @@ void I_BindSoundVariables(void)
     M_BindVariable("snd_samplerate",    &snd_samplerate);
     M_BindVariable("snd_cachesize",     &snd_cachesize);
 
-#ifdef FEATURE_SOUND
+#if defined(FEATURE_SOUND) && !defined(FEATURE_SOUND_GODOT)
     M_BindVariable("use_libsamplerate",   &use_libsamplerate);
     M_BindVariable("libsamplerate_scale", &libsamplerate_scale);
 #endif
diff --git a/doomgeneric/i_swap.h b/doomgeneric/i_swap.h
index 11135c0..189eb82 100644
--- a/doomgeneric/i_swap.h
+++ b/doomgeneric/i_swap.h
@@ -23,7 +23,7 @@
 #ifdef FEATURE_SOUND
 
 
-#ifdef __DJGPP__
+#if defined(__DJGPP__) || defined(FEATURE_SOUND_GODOT)
 
 
 #define SHORT(x)  ((signed short) (x))
